crates/syntax/src/ast/node_ext.rs

```diff
@@ -572,6 +572,16 @@ impl ast::Item {
     }
 }
 
+impl ast::Type {
+    pub fn generic_arg_list(&self) -> Option<ast::GenericArgList> {
+        if let ast::Type::PathType(path_type) = self {
+            path_type.path()?.segment()?.generic_arg_list()
+        } else {
+            None
+        }
+    }
```

Comment:
There maybe more than one generic arg list in a type, `Qux<(Foo<A, B>, Bar<C, D>>)` for example, also generic args might apear in other spots like `[T]`, `(T, U, V)` etc.

Reply:
Thank you for pointing this out! I hadn't considered this situation before.

These situations might make the analysis a bit more complex; I need to reorganize my current approach and think about how to elegantly address this issue. ❤️